### ISSUE TITLE
free(): invalid pointer Aborted (core dumped)

### ISSUE THREAD

**Issue by alexgleason** [DjDeveloperr/skia_canvas#68]
This is basically the same issue as https://github.com/DjDeveloperr/skia_canvas/issues/67 but I fixed the part that was user error (I hope), now I'm seeing this:

```
root@ubuntu-s-1vcpu-512mb-10gb-nyc1-01:~# deno run -A --unstable-ffi captcha.ts
free(): invalid pointer
Aborted (core dumped)
```

Reproduction:

```ts
import { createCaptcha } from 'npm:captcha-canvas@4.0.0';
import { createCanvas } from 'jsr:@gfx/canvas@0.5.6';

const canvas = createCanvas(300, 100);
const ctx = canvas.getContext('2d') as unknown as CanvasRenderingContext2D;

const { text } = createCaptcha({ ctx });

const uri = canvas.toDataURL();

console.log(text);
console.log(uri);
```

I tried to look at the core dump and it's a 548MB ELF file.

**Comment by DjDeveloperr** [comment]
Thanks for reporting this, I've identified the bug in native code. A local structure's pointer was stored in context state which is wrong. I will release a fix ASAP.

**Comment by DjDeveloperr** [comment]
Fixed in 0.5.7
